src/NEPTypes.jl

```diff
@@ -1327,4 +1328,59 @@ Returns true/false if the NEP is sparse (if compute_Mder() returns sparse)
         end
         return z
     end
-end
+
+
+
+    ########## THE FOLLOWING TYPE DerSPMF AND FUNCTION WILL BE DOCUMENTED AND THE CODE BETTER ORGANIZED ##########
+    struct DerSPMF{T<:AbstractMatrix,FDtype,TT<:Number} <: AbstractSPMF{T}
+        spmf::SPMF_NEP{T}
+        fD::Matrix{FDtype}
+        σ::TT
+    end
+
+    # implement all the functions
+    function size(nep::DerSPMF)
+        return size(nep.spmf)
+    end
+    function size(nep::DerSPMF,dim)
+        return size(nep.spmf,dim)
+    end
+    function get_fv(nep::DerSPMF)
+        return get_fv(nep.spmf)
+    end
+    function get_Av(nep::DerSPMF)
+        return get_Av(nep.spmf)
+    end
+
+    ## one constructor takes spmf as input and compute the derivatives
+    function DerSPMF(spmf::SPMF_NEP,σ::Number,m::Int)
```

Comment:
Can't `SPMF_NEP` be replaced by `AbstractSPMF` and use the corresponding get-functions?